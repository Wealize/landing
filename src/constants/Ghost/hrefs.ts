export const CLIENT_STORY_SECTION = 'client-stories'
export const NEWS_ARTICLES_SECTION = 'news-articles'
