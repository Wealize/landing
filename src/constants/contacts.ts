export const GET_IN_TOUCH_HREF = 'https://webforms.pipedrive.com/f/1FSsqSDEX4bH5PDGGiP41BaFyRrkLlEXcIiCLyfIm0aawo27IhbqMpPjWEh8HYEIr'
