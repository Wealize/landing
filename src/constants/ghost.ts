export const API_URL = 'https://wlz-blog.herokuapp.com'
