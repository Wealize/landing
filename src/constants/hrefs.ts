export const HOME_HREF = '/'
export const PROFILE_HREF = '/profile'
export const PRIVACY_HREF = '/privacy'
export const LEGAL_HREF = '/legal'
export const COOKIES_POLICY_HREF = '/cookies-policy'
export const CDTI_HREF = '/grants/cdti'
export const INNOVATION_SME_HREF = '/grants/innovative-sme'
export const NEWS_ROOM_HREF = '/news'
