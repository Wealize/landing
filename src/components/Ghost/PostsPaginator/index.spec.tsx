import React from 'react'
import { shallow } from 'enzyme'
import { Pagination } from '@tryghost/content-api'
import Link from 'next/link'

import { NEWS_ROOM_HREF } from '../../../constants/hrefs'

import GhostPostsPaginator from './index'

describe('<GhostPostsPaginator />', () => {
  const ghostMetaPaginationData: Pagination = {
    page: 1,
    limit: 15,
    pages: 3,
    total: 15,
    next: 2,
    prev: null
  }
  const wrapper = shallow(<GhostPostsPaginator metaPagination={ghostMetaPaginationData} />)

  it('wrapper: should render', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render content correctly', () => {
    expect(wrapper.find(Link).length).toEqual(3)
    expect(wrapper.find('.pagination-page-item.current').length).toEqual(1)
    expect(wrapper.find(Link).at(0).prop('href')).toContain(`${NEWS_ROOM_HREF}?page=1`)
    expect(wrapper.find(Link).at(1).prop('href')).toContain(`${NEWS_ROOM_HREF}?page=2`)
    expect(wrapper.find(Link).at(2).prop('href')).toContain(`${NEWS_ROOM_HREF}?page=3`)
  })
})
