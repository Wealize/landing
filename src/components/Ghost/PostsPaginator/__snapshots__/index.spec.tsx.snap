// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GhostPostsPaginator /> wrapper: should render 1`] = `
<styles__Container>
  <Link
    href="/news?page=1"
    key="0"
    passHref={true}
  >
    <styles__PaginationPage
      className="pagination-page-item current"
    >
      1
    </styles__PaginationPage>
  </Link>
  <Link
    href="/news?page=2"
    key="1"
    passHref={true}
  >
    <styles__PaginationPage
      className="pagination-page-item"
    >
      2
    </styles__PaginationPage>
  </Link>
  <Link
    href="/news?page=3"
    key="2"
    passHref={true}
  >
    <styles__PaginationPage
      className="pagination-page-item"
    >
      3
    </styles__PaginationPage>
  </Link>
</styles__Container>
`;
